pkg/virt-controller/services/template.go

```diff
@@ -1586,12 +1589,29 @@ func copyProbe(probe *v1.Probe) *k8sv1.Probe {
 		SuccessThreshold:    probe.SuccessThreshold,
 		FailureThreshold:    probe.FailureThreshold,
 		Handler: k8sv1.Handler{
+			Exec:      probe.Exec,
 			HTTPGet:   probe.HTTPGet,
 			TCPSocket: probe.TCPSocket,
 		},
 	}
 }
 
+func wrapExecProbeWithVirtProbe(vmi *v1.VirtualMachineInstance, probe *k8sv1.Probe) {
+	if probe == nil || probe.Handler.Exec == nil {
+		return
+	}
+
+	originalCommand := probe.Handler.Exec.Command
+	if len(originalCommand) < 1 {
+		return
+	}
+
+	wrappedCommand := []string{"virt-probe", "--domainName", api.VMINamespaceKeyFunc(vmi), "--command", originalCommand[0]}
+	wrappedCommand = append(wrappedCommand, originalCommand[1:]...)
+
```

Comment:
do we know how much memory this process uses? I think we'll need to add memory overhead for this execution

Reply:
I haven't checked, but would expect it to be very small as it's just a single gRPC call per probe run.
Don't know how tight the resource constraints are and how much is allocated for the container compared to the VM.

Reply:
@davidvossel we could also use conditions instead. Then our controlplane could talk to the respective virt-launchers and run the probes from the outside and update the custom readiness condition:
https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate

Reply:
for now, i think this is as simple as updating the `getMemoryOverhead()` function in template.go. For every readiness/liveliness probe that calls this virt-probe wrap, we need to add some overhead.